site/tests/app/controllers/grading/SimpleGraderControllerTester.php

```diff
@@ -0,0 +1,126 @@
+            ],
+            $response
+        );
+    }
+
+    public function testSaveMissingUserId() {
+        $_REQUEST['user_id'] = null;
+        $core = $this->createMockCore();
+        $controller = new SimpleGraderController($core);
+        $response = $controller->save();
+        $this->assertEquals(
+            [
+                'status' => 'fail',
+                'message' => 'Did not pass in g_id or user_id'
+            ],
+            $response
+        );
+    }
+
+    public function testSaveWrongCsrfToken() {
+        $_REQUEST['g_id'] = 'test';
+        $_REQUEST['user_id'] = 'test';
+        $_REQUEST['csrf_token'] = 'test';
+        $core = $this->createMockCore(['csrf_token' => false]);
+        $controller = new SimpleGraderController($core);
+        $response = $controller->save();
+        $this->assertEquals(
+            [
+                'status' => 'fail',
+                'message' => 'Invalid CSRF token'
+            ],
+            $response
+        );
+    }
+
+    public function testSaveInvalidGradeable() {
+        $_REQUEST['g_id'] = 'test';
+        $_REQUEST['user_id'] = 'test';
+        $_REQUEST['csrf_token'] = 'test';
+        $core = $this->createMockCore(['csrf_token' => true], ['no_user' => true], ['getGradeableConfig' => null]);
+        $controller = new SimpleGraderController($core);
+        $response = $controller->save();
+        $this->assertEquals(
+            [
+                'status' => 'fail',
+                'message' => 'Invalid gradeable ID'
+            ],
+            $response
+        );
+    }
+
+    public function testSaveInvalidUser() {
+        $_REQUEST['g_id'] = 'test';
+        $_REQUEST['user_id'] = 'test';
+        $_REQUEST['csrf_token'] = 'test';
+        $gradeable = $this->createMockModel(Gradeable::class);
+        $core = $this->createMockCore(['csrf_token' => true], ['no_user' => $gradeable], ['getGradeableConfig' => true]);
+        $controller = new SimpleGraderController($core);
+        $response = $controller->save();
+        $this->assertEquals(
+            [
+                'status' => 'fail',
+                'message' => 'Invalid user ID'
+            ],
+            $response
+        );
+    }
+
+    public function testSaveMissingScores() {
+        $_REQUEST['g_id'] = 'test';
+        $_REQUEST['user_id'] = 'test';
+        $_REQUEST['csrf_token'] = 'test';
+        $_POST['scores'] = '';
+        $user = $this->createMockModel(User::class);
+        $gradeable = $this->createMockModel(Gradeable::class);
+        $core = $this->createMockCore(['csrf_token' => true], [], ['getGradeableConfig' => $gradeable, 'getUserById' => $user]);
+        $controller = new SimpleGraderController($core);
+        $response = $controller->save();
+        $this->assertEquals(
+            [
+                'status' => 'fail',
+                'message' => "Didn't submit any scores"
+            ],
+            $response
+        );
+    }
+
```

Comment:
I think the `testSaveNoAccess()` isn't setting the configuration for a user correctly, you can explicitly set the access levels of the mock user you're creating in `createMockCore`, you can do this by setting the configuration of the second param, `$user_config` or the fourth param, `$access`. I think setting setting `$access` to match a user without permission would make more sense for this test since the actual controller uses the `getAccess()` function in the `core` class.